src/main/rendering/rayTracing/BasicRayTracer.java

```diff
@@ -20,13 +22,15 @@
 	 */
 	public BasicRayTracer(Scene scene) {
 		super(scene);
-		// TODO: implement
+		// TODO: implement. Is anything else necessary here?
 	}
 
 	@Override
 	public Colour trace(Ray ray) {
-		// TODO: implement
-		return null;
+		List<Point> intersections = scene.geometries.intersect(ray);
+		if (intersections.isEmpty()) {
+			return scene.background;
+		} 
+		return scene.ambient.colour;
```

Comment:
ternary op is cool, this wld be a nice place to use it lol

Reply:
Why do you think they're cool?